scripts/evaluate_similarity.py

```diff
@@ -18,19 +18,33 @@ def parse_args(args):
         help="Similarity algorithm to use",
         choices=["lsi", "neighbors_tfidf", "neighbors_tfidf_bigrams", "word2vec_wmd"],
     )
+    parser.add_argument(
+        "--disable-url-cleanup",
+        help="Don't cleanup urls when training the similarity model",
+        dest="cleanup_urls",
+        default=True,
+        action="store_false",
+    )
     return parser.parse_args(args)
 
 
 def main(args):
     if args.algorithm == "lsi":
-        model = LSISimilarity()
+        model_creator = LSISimilarity
     elif args.algorithm == "neighbors_tfidf":
-        model = NeighborsSimilarity()
+        model_creator = NeighborsSimilarity
     elif args.algorithm == "neighbors_tfidf_bigrams":
-        model = NeighborsSimilarity(vectorizer=TfidfVectorizer(ngram_range=(1, 2)))
+
+        def model_creator(**kwargs):
```

Comment:
Your latest change doesn't really make a difference, you still have a model_creator function where you are hardcoding the cleanup_urls parameter, so whenever we need to add a new parameter we need to update this function (just like before!).
You should instead add vectorizer to kwargs, and then call NeighborsSimilarity with kwargs.

Reply:
Ack sorry, I'm still a bit confused. If I understand correctly; I should create `kwargs` with `cleanup_urls`, then if the algorithm is `neighbors_tfidf_bigrams`, add the vectorizer argument. Finally, create the model by passing in `kwargs` to `model_creator`; this method would mean I don't have the custom `model_creator` function defined in the `neighbors_tfidf_bigrams` option. Or were you suggesting a different method?

Reply:
@cklyyung the call at line 47 will stay what it is (so it will pass kwargs with `cleanup_urls` to the `model_creator` function at line 38). Then in the `model_creator` function you will update kwargs to add the `vectorizer` parameter and call NeighborsSimilarity with the kwargs.